Medication visualization

jessica-oflanagan opened the pull request:
This pull request fixes some of the awkward spacing in the medication section of the app, when a note is either open or closed.  Ideally, it should be more clear in either view which information belongs to a medication (range-chart, medication change, date prescribed etc) as previously it could be a little unclear.  

Possible Issues: The medication change looks a bit odd in a smaller window (it goes directly under the name of the medication).

**Running the Application**

- [x] Manually tested in Chrome
- [x] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- Cheat sheet is updated
- Demo script is updated 
- Documentation on Wiki has been updated 
- Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- Added Enzyme-UI tests for slim mode 
- Added Enzyme-UI tests for full mode
- Added backend tests for new functionality added
- Added note parser examples to test new functionality


## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code


## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

jafeltra commented on src/summary/MedicationRangeChartVisualizer.jsx:
```diff
@@ -183,91 +246,113 @@ class MedicationRangeChartVisualizer extends Component {
-                            <Col sm={numColsChart}>
+        return (
+            <div key={i} className="medication-chart-item" ref={(parent) => { this.parent = parent }}>
+                <Grid className="FullApp-content" fluid>
```
Is the "FullApp-content" class needed here? I think it's used in other places and wanted to make sure we needed it here as well so that we don't end up with unexpected changes if the class changes.

jafeltra commented on src/summary/MedicationRangeChartVisualizer.jsx:
```diff
@@ -183,91 +246,113 @@ class MedicationRangeChartVisualizer extends Component {
+
+    return (
+        <div key={i} className="medication-chart-item" ref={(parent) => { this.parent = parent }}>
+            <Grid className="FullApp-content" fluid>
```
(same question as above about FullApp-content)

jafeltra commented on src/summary/RangeChart.jsx:
```diff
@@ -75,33 +70,54 @@ class RangeChart extends Component {
-                {/*Data point for the value*/}
-                {svgForDataPoint}
-            </svg>
+            <div>
```
It seems like the svg's here are fairly similar for the wide and not wide cases. Is it possible to maybe put the changes into a variable so we don't have the duplicate svg code? If you already tried that and it doesn't work, that's fine!

jafeltra commented:
The spacing for the medication charts looks really good in all 6 cases that Nicole described in the Jira task. I checked on IE and Chrome and both look good. Nice work!

I added a few comments in the code with a few questions, but nothing major. Besides that, everything looks good to me!

jessica-oflanagan commented:
Just removed the unnecessary code in RangeChart.jsx -- added a variable for the dimensions of the viewbox so that the SVG doesn't need to be entered twice. Also removed the FullApp-content className from both the wide and narrow view in MedicationRangeChartVisualizer.

jafeltra commented:
The changes you made look good to me!

lucyxubroad commented:
It looks really good in IE too!
